### PR TITLE
Bug: Nexus Image Version Must be Semver

### PR DESCRIPTION
**Pull request by jordlay**
---
- Added semver regex from [semver](https://semver.org/)
- Added semver validation to nexus image input config
- Moved split image path into utils

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/aosm/azext_aosm/configuration_models/onboarding_vnf_input_config.py [hunk 1]
```diff
@@ -231,6 +231,11 @@ def validate(self):
             raise ValidationError("arm_template must be set")
         if not self.images:
             raise ValidationError("You must include at least one image")
+        for image in self.images:
+            (_, _, version) = split_image_path(image)
+            if not is_semver(version):
+                raise ValidationError(f"{image} has invalid version '{version}'.\n"
+                                      "Allowed formats are major.minor.patch")
```
Comment (Cyclam) [inline src/aosm/azext_aosm/configuration_models/onboarding_vnf_input_config.py:238]: This is not what the semver regex matches. We should be clear what we allow. Do we actually support semver, or must it be `<int>.<int>.<int>`.

Fixing this involves updating the regex and/or this error message.

File: src/aosm/azext_aosm/common/utils.py [hunk 2]
```diff
@@ -130,5 +130,11 @@ def split_image_path(image) -> "tuple[str, str, str]":
...
     return (source_acr_registry, name, version)
 
-def is_semver(string):
-    return re.match(SEMVER_REGEX, string) is not None
+def is_valid_nexus_image_version(string):
+    """Check if image version is valid.
+    
+    This is based on validation in pez repo.
+    It requires the image version to be major.minor.path,
+    but does enforce full semver validation.
```
Comment (Cyclam) [inline src/aosm/azext_aosm/common/utils.py:138]: Not sure what line 138 means. Maybe it should be "but does not support full semver pre-release naming."?

File: src/aosm/azext_aosm/common/utils.py [hunk 3]
```diff
@@ -130,5 +130,11 @@ def split_image_path(image) -> "tuple[str, str, str]":
     (name, version) = name_and_version.split(":", 2)
     return (source_acr_registry, name, version)
 
-def is_semver(string):
-    return re.match(SEMVER_REGEX, string) is not None
+def is_valid_nexus_image_version(string):
+    """Check if image version is valid.
+    
+    This is based on validation in pez repo.
+    It requires the image version to be major.minor.path,
```
Comment (Cyclam) [inline src/aosm/azext_aosm/common/utils.py:137]: Typo: path > patch.

(Code suggestion has disappeared!)

File: src/aosm/azext_aosm/configuration_models/onboarding_vnf_input_config.py [hunk 4]
```diff
@@ -233,7 +233,7 @@ def validate(self):
             raise ValidationError("You must include at least one image")
         for image in self.images:
             (_, _, version) = split_image_path(image)
-            if not is_semver(version):
+            if not is_valid_nexus_image_version(version):
                 raise ValidationError(f"{image} has invalid version '{version}'.\n"
                                       "Allowed formats are major.minor.patch")
```
Comment (Cyclam) [inline src/aosm/azext_aosm/configuration_models/onboarding_vnf_input_config.py:238]: Suggest changing 
"Allowed formats are major.minor.patch" to
"Allowed format is major.minor.patch"

File: src/aosm/azext_aosm/common/utils.py (hunk 2 above)
Comment (jordlay) [reply src/aosm/azext_aosm/common/utils.py:138]: meant to be "does not" fixing

**Pull request merged**